syncserver/src/lib.rs

```diff
@@ -3,12 +3,243 @@
 
 #[macro_use]
 extern crate slog_scope;
-#[macro_use]
-extern crate validator_derive;
 
-#[macro_use]
-pub mod error;
 pub mod logging;
-pub mod server;
-pub mod tokenserver;
-pub mod web;
+mod middleware;
+
+use std::{error::Error, sync::Arc};
+
+use actix_cors::Cors;
+#[cfg(feature = "syncstorage")]
+use actix_web::http::StatusCode;
+#[cfg(feature = "syncstorage")]
+use actix_web::middleware::errhandlers::ErrorHandlers;
+use actix_web::{dev, http::Method, web, App, HttpServer};
+#[cfg(not(all(feature = "syncstorage", feature = "tokenserver")))]
+use actix_web::{web::ServiceConfig, HttpRequest, HttpResponse};
+use syncserver_settings::Settings;
+
+#[cfg(not(any(feature = "syncstorage", feature = "tokenserver")))]
+compile_error!("At least one of the \"syncstorage\" or \"tokenserver\" features must be enabled for this crate.");
+
+pub struct Server;
+
+#[cfg(all(feature = "syncstorage", not(feature = "tokenserver")))]
+#[macro_export]
+macro_rules! build_app {
+    ($settings: expr) => {
+        App::new()
+            .data(Arc::new($settings.master_secret.clone()))
+            // Middleware is applied LIFO
+            // These will wrap all outbound responses with matching status codes.
+            .wrap(ErrorHandlers::new().handler(
```

Comment:
These middleware registrations should be replaced with the new `wrap_fn` paradigm (we already do this on the main branch, I just didn't get a chance to update it here)